src/rust/engine/src/graph.rs

```diff
@@ -443,6 +445,36 @@ impl InnerGraph {
     try!(f.write_all(b"\n"));
     Ok(())
   }
+
+  pub fn all_digests(&self, core: Arc<Core>) -> Vec<hashing::Digest> {
+    self
+      .nodes
+      .keys()
+      .map(|key| self.entry(key))
```

Comment:
Rather than "find all kets and lookup each entry", you can directly "get all entries" with `self.pg.node_weights_mut`: https://docs.rs/petgraph/0.4.11/petgraph/graph/struct.Graph.html#method.node_weights_mut (not sure why it only has a `_mut` form, but.)

Reply:
Done